validator/sawtooth_validator/journal/publisher.py

```diff
@@ -591,26 +595,30 @@ def _build_candidate_block(self, chain_head):
             else:
                 break
 
-    def on_batch_received(self, batch):
+    def on_batches_received(self, batches):
+        with self._lock:
+            for batch in batches:
+                self._on_batch_received(batch)
+
+    def _on_batch_received(self, batch):
         """
         A new batch is received, send it for validation
         :param batch: the new pending batch
         :return: None
         """
-        with self._lock:
-            self._queued_batch_ids = self._queued_batch_ids[:1]
-            if self._permission_verifier.is_batch_signer_authorized(batch):
-                self._pending_batches.append(batch)
-                self._pending_batch_ids.append(batch.header_signature)
-                self._set_gauge(len(self._pending_batches))
-                # if we are building a block then send schedule it for
-                # execution.
-                if self._candidate_block and \
-                        self._candidate_block.can_add_batch:
-                    self._candidate_block.add_batch(batch)
-            else:
-                LOGGER.debug("Batch has an unauthorized signer. Batch: %s",
-                             batch.header_signature)
+        self._queued_batch_ids = self._queued_batch_ids[:1]
+        if self._permission_verifier.is_batch_signer_authorized(batch):
+            self._pending_batches.append(batch)
```

Comment:
_pending_batches is fiddled with in a few places. I'm a little wary of removing locks from it. 
Have you already assessed risks on these sections:
validator/sawtooth_validator/journal/publisher.py:735

```                    last_batch_index = self._pending_batches.index(last_batch)
                    unsent_batches = \
                        self._pending_batches[last_batch_index + 1:]
                    self._pending_batches = pending_batches + unsent_batches

                    self._pending_batch_gauge.set_value(
                        len(self._pending_batches))
```

And this section
validator/sawtooth_validator/journal/publisher.py:597

```        for batch in self._pending_batches:
            if self._candidate_block.can_add_batch:
                self._candidate_block.add_batch(batch)
            else:
                break
```